### ISSUE TITLE
training model

### ISSUE THREAD

**Issue by lavrenkov-sketch** [ultralytics/yolov5#4391]
## 1.
@glenn-jocher  hi, could you help me, is it possible to continue training the already trained model? how to do it correctly?
## 2.
is it possible to add data to the dataset and continue training? How does it affect the quality of the model?
## 3.
during normal training, hyperparameters are calculated or for them it is still better to calculate using --evolve

**Comment by glenn-jocher** [comment]
@lavrenkov-sketch 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck and let us know if you have any other questions!

**Comment by lavrenkov-sketch** [comment]
@glenn-jocher and if I see that the model is already ready, I can force the optimization ahead of schedule?

**Comment by glenn-jocher** [comment]
@lavrenkov-sketch I don't know what you are asking
